### ISSUE TITLE
Adrenaline::__invoke() not compatible with Adroit::__invoke()

### ISSUE THREAD

**Issue by shochdoerfer** [bitExpert/adrenaline#15]
As it turns out the current implementation of Adrenaline::__invoke() is not  compatible with Adroit::__invoke() as the latter does return the Response and Adrenaline does not. How do we want to address this issue?

**Comment by dropdevcoding** [comment]
Currently Adrenaline inherits from Adroit which imho is not the best approach. Instead I would prefer Adrenaline to make use of Adroit but I couldn't figure out why It didn't work. Maybe it is possible right now due to the Stratigility upgrade and if that worked it would also solve the problem  this issue addresses. So would you mind opening an issue for that? Of course it has to be changed but I don't want Adrenaline to return the response since it is a framework which emits the response whilst Adroit is a middleware.

**Comment by shochdoerfer** [comment]
@dropdevcoding why opening another issue? Isn't this sufficient? ;) Mind working on a proposal for a change?

**Comment by dropdevcoding** [comment]
You're right, sufficient enough. I'd like to work on a PR but unfortunately I won't find the time to before thursday. If that is ok for you, I'll try to propose a solution until friday / saturday

**Comment by shochdoerfer** [comment]
Sure not a problem at all. Thanks for taking care of the issue!
